src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/repo/RepoInfos.java

```diff
@@ -32,11 +31,12 @@ public static RepoInfo getByRepoId(String repoId, QueryRunner q) throws SQLExcep
             + " repos where repoid=?",
         rs -> {
           rs.next();
+          Timestamp expiryDate = rs.getTimestamp("expiryDate");
           return new RepoInfo(
               rs.getString("repoid"),
               rs.getString("taskid"),
               rs.getBoolean("using_testing_version"),
-              new Date(rs.getTimestamp("expiryDate").getTime()),
+              expiryDate != null ? new Date(expiryDate.getTime()) : null,
```

Comment:
Could you check that the jdbc standard guarantees you will get a java null for an sql null - i vaguely remember something about an isNull method or something?

Reply:
https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#getTimestamp(java.lang.String) says it returns NULL if SQL is NULL. I think `wasNull` is only needed for primitives.

Reply:
OK. Cool. Thanks for checking